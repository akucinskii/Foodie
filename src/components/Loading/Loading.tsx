import { useSession } from "next-auth/react";
import React from "react";

type Props = {
  children: React.ReactNode;
};

const Loading = ({ children }: Props) => {
  const session = useSession();

  return !(session.status === "loading") ? (
    <>{children}</>
  ) : (
    <div
      className="
    flex
    h-full
    w-full
    flex-col
    items-center
    justify-center
    pt-20
    text-2xl
    "
    >
      Loading...
    </div>
  );
};

export default Loading;
