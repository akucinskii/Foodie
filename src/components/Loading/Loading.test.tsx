import { render } from "@testing-library/react";
import { Session } from "next-auth";
import { useSession } from "next-auth/react";
import { describe, expect, Mock, test, vi } from "vitest";
import Loading from "./Loading";

const mockSession: Session = {
  user: {
    id: "1",
    name: "Test",
    email: "test@test.test",
    image: "https://testimages.org/img/testimages_screenshot.jpg",
  },
  expires: "1",
};

vi.mock("next-auth/react", () => ({
  signIn: vi.fn(),
  signOut: vi.fn(),
  useSession: vi.fn(),
}));

describe("Loading", () => {
  (useSession as Mock).mockReturnValueOnce({
    data: mockSession,
    status: "authenticated",
  });
  const { container, rerender } = render(<Loading>Children</Loading>);

  test("Should match snapshot", () => {
    (useSession as Mock).mockReturnValueOnce({
      data: mockSession,
      status: "authenticated",
    });

    rerender(<Loading>Children</Loading>);
    expect(container).toMatchSnapshot("Children");
  });

  test("Should match snapshot when loading", () => {
    (useSession as Mock).mockReturnValueOnce({
      data: mockSession,
      status: "loading",
    });

    rerender(<Loading>Children</Loading>);

    expect(container).toMatchSnapshot("loading");
  });
});
