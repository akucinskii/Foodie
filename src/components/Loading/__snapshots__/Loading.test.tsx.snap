// Vitest Snapshot v1

exports[`Loading > Should match snapshot > Children 1`] = `
<div>
  Children
</div>
`;

exports[`Loading > Should match snapshot 1`] = `
<div>
  Children
</div>
`;

exports[`Loading > Should match snapshot when loading > loading 1`] = `
<div>
  <div
    class=" flex h-full w-full flex-col items-center justify-center pt-20 text-2xl "
  >
    Loading...
  </div>
</div>
`;
